File: 169-MajorityElement/169-MajorityElement.cpp
// Last updated: 4/3/2026, 2:01:49 PM
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int freq = 0, ans = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (freq == 0) {
                ans = nums[i];
            }
            if (ans == nums[i]) {
                freq++;
            } else {
                freq--;
            }
        }
        // If the majority element is not always present
        // Check the freq of ans in another loop
        // if(freq > (n/2)) YES else NO
        return ans;
    }
};

File: 169-MajorityElement/169-MajorityElement.py
# Last updated: 4/3/2026, 2:01:45 PM
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count, num = 0, nums[0]
        for i in nums:
            if count == 0:
                num = i
            if i == num:
                count += 1
            else:
                count -= 1
        return num